synapse/config/sso.py

```diff
@@ -51,6 +49,22 @@ def read_config(self, config, **kwargs):
             self.sso_template_dir,
         )
 
+        # The following templates require a public baseurl since they use the
+        # mxc_to_http filter.
+        #
+        # Note that any situation where these templates get used will check that
+        # public_baseurl is set already.
+        self.sso_login_idp_picker_template = None  # type: Optional[jinja2.Template]
+        self.sso_redirect_confirm_template = None  # type: Optional[jinja2.Template]
+        if self.public_baseurl:
+            (
+                self.sso_login_idp_picker_template,
+                self.sso_redirect_confirm_template,
+            ) = self.read_templates(
+                ["sso_login_idp_picker.html", "sso_redirect_confirm.html"],
+                self.sso_template_dir,
+            )
+
```

Comment:
this feels a bit fragile, and obscure. Those templates may be the only ones that currently use `mxc_to_http` by default, but I can see us updating other templates to use it in the future, or people customising other templates and wondering why it mysteriously fails.

it might be better to arrange for `mxc_to_http` to always be defined (so that the template can be loaded), but then raise an exception at render time, with an explanation that `public_baseurl` is required in the error?

Reply:
I agree, I don't love it. A runtime error might make more sense, especially since we do always expect it to be configured when needed.

Reply:
I made this change -- `mxx_to_http` will now raise a runtime error if `public_baseurl` is not set. It isn't expected that this actually happens since we do check in the config the situations where we expect it to be required.